src/pages/api/oauth/accept-consent.ts

```diff
@@ -1,38 +1,49 @@
+import { Session } from '@ory/client'
 import { gql } from 'graphql-request'
-import { NextApiRequest, NextApiResponse } from 'next'
+import type { NextRequest } from 'next/server'
 
 import { createGraphqlFetch } from '@/api/graphql-fetch'
-import { AuthSessionCookie } from '@/auth/auth-session-cookie'
+import { KRATOS_HOST } from '@/auth/kratos-host'
 
 export const config = {
   runtime: 'experimental-edge',
 }
 
-export default async function acceptConsent(
-  req: NextApiRequest,
-  res: NextApiResponse
-) {
-  const { consent_challenge } = req.query
+export default async function acceptConsent(req: NextRequest) {
+  if (!KRATOS_HOST)
+    return new Response('missing env var', {
+      status: 500,
+    })
 
-  const session = AuthSessionCookie.parse(req.cookies)
-  const query = gql`
-    mutation ($input: OauthAcceptInput!) {
-      oauth {
-        acceptConsent(input: $input) {
-          redirectUri
+  try {
+    const { searchParams } = new URL(req.url)
+    const challenge = searchParams.get('consent_challenge')
+
+    const cookie = req.cookies.get('ory_kratos_session') ?? ''
+    const sessionResponse = await fetch(`${KRATOS_HOST}/sessions/whoami`, {
```

Comment:
I'm getting here the following error:
```
TypeError [ERR_INVALID_URL]: Invalid URL
```
it makes sense also. 

Besides that and more important: `kratos.toSession` is just a less verbose wrapper around a request that hits this same endpoint. Let me try again with it and I'll tell you here

Reply:
interesting, in my tests I never got this error.

> Besides that and more important: kratos.toSession is just a less verbose wrapper around a request that hits this same endpoint.

true, I just tried it this way, but `.toSession` should work fine on the server if we forward the cookie.